apps/examples/testcase/le_tc/filesystem/fs_main.c

```diff
@@ -2346,9 +2465,21 @@ static void libc_stdio_ungetc_tc(void)
 	ch1 = fgetc(fp);
 	TC_ASSERT_NEQ_CLEANUP("fgetc", ch1, EOF, fclose(fp));
 
+	/* Negative case with invalid argument, NULL stream. It will return EOF */
+
+	ret = ungetc(64, NULL);
+	TC_ASSERT_EQ_CLEANUP("ungetc", ret, EOF, fclose(fp));
+
 	ret = ungetc(64, fp);
 	TC_ASSERT_NEQ_CLEANUP("ungetc", ret, EOF, fclose(fp));
 
+	/* Negative case with invalid argument. It will return EOF */
+
+#if CONFIG_NUNGET_CHARS > 0
+	fp->fs_nungotten = 4;
+	ret = ungetc(64, fp);
+	TC_ASSERT_EQ_CLEANUP("ungetc", ret, EOF, fclose(fp));
```

Comment:
You set fs_nungotten as invalid value which is out of range for this tc.
But it can bring assert in next tc, fgetc because this value is used as index of fs_ungotten array of which the size is CONFIG_NUNGET_CHARS, 2.
please check this, line 132 in lib_libfread.c